rust/agents/relayer/src/msg/metadata/base.rs

```diff
@@ -45,11 +46,13 @@ pub trait MetadataBuilder {
 
 #[derive(Clone, new)]
 pub struct BaseMetadataBuilder {
-    chain_setup: ChainConf,
-    prover_sync: Arc<RwLock<MerkleTreeBuilder>>,
-    validator_announce: Arc<dyn ValidatorAnnounce>,
-    allow_local_checkpoint_syncers: bool,
-    metrics: Arc<CoreMetrics>,
+    pub chain_setup: ChainConf,
+    pub prover_sync: Arc<RwLock<MerkleTreeBuilder>>,
+    pub validator_announce: Arc<dyn ValidatorAnnounce>,
+    pub allow_local_checkpoint_syncers: bool,
+    pub metrics: Arc<CoreMetrics>,
+    pub depth: u32,
```

Comment:
`depth` represents the depth of the recursion when building metadata.

How would CloneDepthLimit<T> work, exactly? Since we do the cloning in `build`, would we need to implement `MetadataBuilder` for `CloneDepthLimiter<BaseMetadataBuilder>`? That would feel weirder to me.

Reply:
Thinking on it some more it is fine the way it is being done here for now. Just please add some docs to make it more clear in the code.